app/lib/three_scale/jobs.rb

```diff
@@ -43,7 +45,8 @@ def map(tasks)
 
         def deserialize(args)
           hash = normalize_task_args(args)
-          klass, method, arguments = YAML.load(hash[:init_args]) # rubocop:disable Security/YAMLLoad
+          permitted_classes = ActiveRecord::Base.yaml_column_permitted_classes + ThreeScale::Jobs::JOB_CLASSES
+          klass, method, arguments = YAML.load(hash[:init_args], permitted_classes: permitted_classes)
```

Comment:
if we don't load user generated content, then perhaps simple unsafe loading will suffice? If somebody can control the input configuration file, they should be trusted enough not already and can mess up everything in easier ways if they so wish

Reply:
Ehm... are you saying... we neeed to trust the developers?! :scream: 
That's certainly a doubtful practice :grimacing: 

If you prefer to change it to make it simpler, please go for it!

Reply:
Certainly trusting the source of all bugs is unwise. I'm just saying the trust is not effectively expanded :angel: 

Check #3905